src/mqttcm_connect.c

```diff
@@ -414,6 +416,49 @@ int validateForMqttInit()
 	return 0;
 }
 
+void custom_log_callback(struct mosquitto *mosq, void *userdata, int level, const char *message)
+{
+    // Control logging based on levels
+    if (level == MOSQ_LOG_DEBUG) {
+        // Log debug messages
+        MqttCMDebug("Debug log message: %s\n", message);
+    } else if (level == MOSQ_LOG_INFO) {
+        // Log informational messages
+        MqttCMDebug("Info log message: %s\n", message);
+    } else if (level == MOSQ_LOG_NOTICE) {
+        // Log notice messages
+        MqttCMDebug("Notice log message: %s\n", message);
+    } else if (level == MOSQ_LOG_WARNING) {
+        // Log warning messages
+        MqttCMDebug("Warning log message: %s\n", message);
+    } else if (level == MOSQ_LOG_ERR) {
+        // Log error messages
+        MqttCMError("Error log message: %s\n", message);
+    }
+}
+
+int password_callback(char *buf, int size, int rwflag, void *userdata)
+{
+	MqttCMInfo("Inside Password_callback\n");
+	char *passphrase = NULL;
+	int len =0;
+
+	get_from_file("private_key_passwd=", &passphrase, "/dev/stdin");
+
+	if( !passphrase )
+	{
+		MqttCMError("Passkey not Found\n");
+	}
+	else
+	{
+		strncpy(buf, passphrase, size);
```

Comment:
where buf is getting allocated ? buf NULL check is needed .
Also, add the unit test for buf Null case

Reply:
Added size and buf check, added unit test cases for null check and size